### PR TITLE
Cluster-autoscaler: move ToBeDeleted taint functions to utils

### PR DESCRIPTION
**Pull request by mwielgus**
Minor cleanup. No much of new code.

cc: @jszczepkowski @fgrzadkowski

### CODE REVIEW

File: cluster-autoscaler/scale_down.go [hunk 1]
```diff
@@ -335,96 +334,19 @@ func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface
...
-		return false, err
-	}
-	if node.Annotations == nil {
-		node.Annotations = make(map[string]string)
-	}
-	node.Annotations[apiv1.TaintsAnnotationKey] = string(taintsJson)
-	return true, nil
-}
-
 // cleanToBeDeleted clean ToBeDeleted taints.
```
Comment (jszczepkowski) [inline cluster-autoscaler/scale_down.go]: s/clean/cleans

Comment (jszczepkowski) [inline cluster-autoscaler/scale_down.go]: s/taints/taint

File: cluster-autoscaler/cluster_autoscaler.go [hunk 2]
```diff
@@ -245,10 +245,7 @@ func run(_ <-chan struct{}) {
 				}
 
 				// CA can die at any time. Removing taints that might have been left from the previous run.
-				if err := cleanToBeDeleted(nodes, kubeClient, autoscalingContext.Recorder); err != nil {
-					glog.Warningf("Failed to clean ToBeDeleted information: %v", err)
-					continue
-				}
+				cleanToBeDeleted(nodes, kubeClient, autoscalingContext.Recorder)
```
Comment (jszczepkowski) [inline cluster-autoscaler/cluster_autoscaler.go]: What if cleanToBeDeleted internally failed? Why are you changing the semantic here? Is it correct?

File: cluster-autoscaler/utils/deletetaint/delete.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		return false, err
+	}
+	if node.Annotations == nil {
+		node.Annotations = make(map[string]string)
+	}
+	node.Annotations[apiv1.TaintsAnnotationKey] = string(taintsJson)
+	return true, nil
+}
+
+// HasToBeDeletedTaint return true if ToBeDeleted taint is applied on the node.
```
Comment (jszczepkowski) [inline cluster-autoscaler/utils/deletetaint/delete.go]: s/return/returns

File: cluster-autoscaler/scale_down.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/scale_down.go]: DONE.

File: cluster-autoscaler/cluster_autoscaler.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: We will try to continue anyway. One node with broken taints should not stop CA completely.

**Pull request merged**